content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,288 @@
+---
+title: DRA Resource Drivers
+description: Resource drivers provide non-trivial allocation logic and management for devices or resources that require vendor-specific or just complex setup, such as GPUs, NICs, FPGAs, etc.
+content_type: concept
+weight: 10
+---
+
+<!-- overview -->
+{{< feature-state for_k8s_version="v1.26" state="alpha" >}}
+
+Kubernetes provides a
+[Dynamic Resource Allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) (DRA)
+mechanism
+that can be leveraged to provide more complex hardware resources to workloads with custom resource
+accounting.
+
+Similarly to {{< glossary_tooltip term_id="device-plugin" text="device plugins">}}, instead of
+customizing the code for Kubernetes itself, vendors can implement a _resource driver_ that you deploy
+into the cluster to account for and control the allocation of GPUs, high-performance NICs, FPGAs,
+InfiniBand adapters, and other similar computing resources that may require vendor specific
+initialization and setup.
+
+With device plugins, the scheduler was given a trivial, numerical representation of the resources
+available on node for consideration during scheduling as an extended resource.
+
+With DRA, the scheduler is offloading the task of allocating and accounting for non-native resources
+to the resource driver, which manages such resources in the cluster.
+
+Resource driver consists of two main components:
+
+- a _controller_ (one per cluster), manages hardware resources allocation for
+  {{< glossary_tooltip term_id="resource-claim" text="ResourceClaims">}}
+- _kubelet-plugin_ (one per node with resource), that:
+  - discovers the supported hardware
+  - announces the discovered hardware to the Resource Driver controller
+  - prepares the hardware allocated for Pod when Pod comes to the Kubelet
+  - unprepares the hardware allocated for a Pod when the Pod has reached final state or is being deleted.
+
+There are two common ways to implement communication between the controller and a kubelet plugin:
+
+- through {{< glossary_tooltip term_id="CustomResourceDefinition" text="CRD">}} objects provided by
+  and designed for the resource driver
+- through a ResourceHandle which is a part of an `AllocationResult` provided by the controller in case of
+  successful allocation
+
+General recommendations:
+
+- resource driver name pattern: `<HW type>.resource.<companyname>.<companydomain>`. For example,
+  gpu.resource.example.com
+
+<!-- body -->
+
+## Resource driver registration
+
+Controller and kubelet-plugin registration can be handled with the DRA _helper code_ that implements
+common for all resource drivers functionality.
```

Comment:
Drop the second "functionality"?
```suggestion
common for all resource drivers.
```

Reply:
If we apply the suggestion that makes last "functionality" a second one, I'll remove it. At the moment it is the only one.